### DISCUSSION TITLE
IBM for the Thin Nozzles - How to define nobjmax

### DISCUSSION THREAD

**Discussion by Fahimeh-Goodarzi** [xcompact3d/Incompact3d#296] [Q&A] [answered]
Dear all
I’m encountering while running simulations using Xcompact3D (v4.1). My setup is two round jets ejected from thin cylindrical nozzles extending from the x = 0 plane (a wall). I am using the immersed boundary method for the nozzles. **I am uncertain about how to define nobjmax for this case**, as I noticed in the examples for a single cylinder nobjmax is set to 2.
In our case, with two nozzles, I first set nobjmax = 2 but my simulation failed, however I increase the nobjmax to 4(uncertain is it correct or not) and my simulation converged. 
Additionally, for thin geometries like the ones I’m working with, which IBM method would you recommend for optimal handling? I simply opted for the cubic spline (iibm = 3), but it seems I may have to significantly refine my mesh in y and z to have my simulation converge. When I remove the nozzles (jets emanating from orifices), the simulation converges well.
Any additional insight on the IBM configuration would be greatly appreciated.

&ibmstuff
!================

imove        = 0        ! (0: stationary objects, 1: moving objects)
ubcx         = 0.00     ! Prescribed velocity (ux)
ubcy         = 0.00     ! Prescribed velocity (uy)
ubcz         = 0.00     ! Prescribed velocity (uz)
nraf         = 10       ! Level of refinement (iibm = 2 or 4) to find the surface of the immersed object
nobjmax      = 2        ! Number of immersed objects (DO NOT USE ZERO OBJECTS)
npif         = 2        ! Number of points for the reconstruction (npif = 1-3, recomended: 2)
izap         = 1        ! How many points to skip for reconstruction (izap = 0-3, reccomended: 1)
ianal        = 0        ! Boundary position approximation? (0: refinement approximation, 1: analytical)

**Comment by slaizet** [comment, marked as answer]
Hi,
NOBJMAX can be defined as the number of time you will cross a fluid/solid and solid/fluid interface in each spatial direction. To use reconstruction approaches for the IBM you will need at least 5/6 points in each spatial direction for a solid region.
I hope this helps.
Best,
Sylvain

**Reply by Fahimeh-Goodarzi** [reply]
thank you very much for your help.
